File: Merge Intervals.js
var merge = function(intervals) {
    intervals.sort((a,b) => a[0] - b[0]);
  const res = [];
  for(let interval of intervals) {
      const prev = res.at(-1);//index starts at the back
      if (!res.length || prev[1] < interval[0]) {
          res.push(interval);
      } else {
          prev[1] = Math.max(prev[1], interval[1])
      }

  }
  return res;
};

File: 56-merge-intervals/merge-intervals.py
class Solution(object):
    def merge(self, intervals):
        
        if not intervals:
            return []
        intervals.sort(key= lambda x:x[0])
        res  = [intervals[0]]
        for i in range(1,len(intervals)):
            prev = res[-1]
            current = intervals[i]
            if prev[1] >= current[0]:
                prev[1] = max(current[1], prev[1])
                
            else:
                res.append(current)
        return res


      
        
      

       
        